File: 진우/13weeks/18258. 큐 2/큐 2.cc
#include <iostream>
#include <queue>

using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    cin >> N;   // 명령어 개수 입력

    queue<int> q;  // 정수를 저장할 큐 선언 (FIFO)

    string cmd;
    for (int i = 0; i < N; i++) {
        cin >> cmd;   // 명령 입력

        // 정수 X를 큐에 넣기
        if (cmd == "push") {
            int X;
            cin >> X;
            q.push(X);
        }
        // 큐의 front 요소 출력 후 제거
        else if (cmd == "pop") {
            if (!q.empty()) {
                cout << q.front() << "\n";
                q.pop();
            }
            else cout << -1 << "\n";  // 큐가 비었으면 -1 출력
        }
        // 큐의 현재 크기 출력
        else if (cmd == "size") {
            cout << q.size() << "\n";
        }
        // 큐가 비었으면 1, 아니면 0 출력
        else if (cmd == "empty") {
            if (q.empty()) cout << 1 << "\n";
            else cout << 0 << "\n";
        }
        // 큐의 front 요소 출력 (제거 X)
        else if (cmd == "front") {
            if (!q.empty()) cout << q.front() << "\n";
            else cout << -1 << "\n";
        }
        // 큐의 back 요소 출력 (제거 X)
        else if (cmd == "back") {
            if (!q.empty()) cout << q.back() << "\n";
            else cout << -1 << "\n";
        }
    }

    return 0;
}

File: 진우/13weeks/18258. 큐 2/큐 2.py
import sys
input = sys.stdin.readline   # 빠른 입력
from collections import deque

N = int(input())  # 명령어 개수 입력

q = deque()       # 정수를 저장할 큐 (FIFO)

for _ in range(N):
    cmd = input().split()  # 명령 입력 (push X 형태 대비 split 사용)

    # push X : 정수 X를 큐에 삽입
    if cmd[0] == "push":
        q.append(int(cmd[1]))

    # pop : 큐의 첫 원소 출력 후 제거
    elif cmd[0] == "pop":
        if q:
            print(q.popleft())
        else:
            print(-1)

    # size : 큐 크기 출력
    elif cmd[0] == "size":
        print(len(q))

    # empty : 큐가 비었으면 1, 아니면 0 출력
    elif cmd[0] == "empty":
        if q:
            print(0)
        else:
            print(1)
            
    # front : 큐의 첫 원소 출력 (제거 X)
    elif cmd[0] == "front":
        if q:
            print(q[0])
        else:
            print(-1)

    # back : 큐의 마지막 원소 출력 (제거 X)
    elif cmd[0] == "back":
        if q:
            print(q[-1])
        else:
            print(-1)
